src/cgutils.cpp

```diff
@@ -2670,6 +2704,7 @@ static jl_cgval_t emit_getfield_knownidx(jl_codectx_t &ctx, const jl_cgval_t &st
             if (first_ptr)
                 null_pointer_check(ctx, first_ptr, nullcheck);
         }
+        setName(ctx.emission_context, fldv, [&]() { return (get_objname() + "." + get_fieldname(jt, idx)).str(); });
```

Comment:
This is asserting for some reason .

Reply:
That assertion is my way of checking that we don't overwrite argument names. If we're doing that, it implies we're also overwriting other variables which might have names with more semantic meaning, and therefore we should call `setName` more conservatively.